File: node.js/src/original/domain/books/events/bookPublishedEvent.ts
import { DomainEvent } from '../../../infrastructure/events';
import { Author, BookId, ISBN, Title } from '../entities';

export type BookPublishedEvent = DomainEvent<
  'BookPublishedEvent',
  {
    bookId: BookId;
    isbn: ISBN;
    title: Title;
    author: Author;
  }
>;

File: java/src/main/java/io/eventdriven/slimdownaggregates/original/domain/books/events/BookPublishedEvent.java
package io.eventdriven.slimdownaggregates.original.domain.books.events;

import io.eventdriven.slimdownaggregates.original.domain.books.entities.*;
import io.eventdriven.slimdownaggregates.original.infrastructure.events.DomainEvent;

public record BookPublishedEvent(
  BookId bookId,
  ISBN isbn,
  Title title,
  Author author
) implements DomainEvent {
}
